openff/bespokefit/schema/fitting.py

```diff
@@ -81,6 +83,19 @@ class BaseOptimizationSchema(SchemaBase, abc.ABC):
         "stage.",
     )
 
+    @property
+    def target_smirks(self) -> List[SMIRKSType]:
+        """Returns a list of the target smirks types based on the selected hyper parameters.
+        Used to determine which parameters should be fit.
+        """
+        return list(
+            {
+                SMIRKSType(parameter.type)
+                for stage in self.stages
+                for parameter in stage.parameter_hyperparameters
+            }
+        )
+
```

Comment:
It may be better to have this live on the bespoke schema as I don't think we use this in the general fits? We can always easily move it back here to the parent class later if we need it without breaking the API.